tests_metricflow/snapshots/test_conversion_metric_rendering.py/SqlQueryPlan/DuckDB/test_conversion_metric_with_categorical_filter__plan0_optimized.sql

```diff
@@ -18,20 +18,28 @@ FROM (
       , visit__referrer_id
       , SUM(visits) AS visits
     FROM (
-      -- Read Elements From Semantic Model 'visits_source'
-      -- Metric Time Dimension 'ds'
+      -- Constrain Output with WHERE
       -- Pass Only Elements: ['visits', 'visit__referrer_id', 'metric_time__day']
       SELECT
-        DATE_TRUNC('day', ds) AS metric_time__day
-        , referrer_id AS visit__referrer_id
-        , 1 AS visits
-      FROM ***************************.fct_visits visits_source_src_28000
-    ) subq_18
+        metric_time__day
+        , visit__referrer_id
+        , visits
+      FROM (
+        -- Read Elements From Semantic Model 'visits_source'
+        -- Metric Time Dimension 'ds'
+        SELECT
```

Comment:
I'm not 100% sure. The basic subquery reducer skips any subquery with a where expression, but the rewriting subquery reducer should do this collapsing at least for the inner where query. It's a bit odd that it doesn't, and I haven't figured out why not. My initial theory was that this is due to the nested where constraint subqueries, but if that's the case I don't understand why it wasn't collapsed in the first place. It may be an interaction between the join and filter subqueries - renaming things is harder when joins get involved.